### PR TITLE
factoring out refresh()

### PR DESCRIPTION
**Pull request by jeremybmerrill**
I’ve factored out `refresh()` in favor of some component lifecycle stuff

all the non-refresh-related tests pass (and I wrote some new ones), but I’d love if, before we merge, you take a look at the functionality and make sure I didn’t make any subtle UI changes by accident
also your thoughts on how I approached the problem would be welcome too :)

but no rush, of course

### CODE REVIEW

File: backend/server/public/src/components/adlist.jsx [hunk 1]
```diff
@@ -5,56 +5,85 @@ import Term from "components/term.jsx";
...
-        <legend>{t("search_terms")}</legend>
-        <ul>
-          {["Trump", "Obama", "Hillary", "Mueller", "Health", "Taxes"].map(
-            term => <Term key={term} search={search} term={term} />
+class AdListUnconnected extends React.Component {
+  componentDidMount() {
+    this.props.deserialize(); // gets params from the URL and dispatches the relevant actions, which'll cause a DidUpdate, and then a getAds
+  }
+
+  // if page or search changes, then we'll update the ads from the server.
```
Comment (jeremybmerrill) [inline backend/server/public/src/components/adlist.jsx]: annotate this better

**Comment by thejefflarson** [comment]
Ok I added some that works around the lifecycle event stuff, @jeremybmerrill can you take a look and make sure it all works? Also the tests for adlist and ads are lacking with these changes, can we try and get our coverage back up? Other than those two things this is good to go!

**Comment by jeremybmerrill** [comment]
Okay, I learned a bunch about mocking functions and now those two files have 100% statement coverage. Things seem to work right too. Take a look?

**Comment by thejefflarson** [comment]
Looks good to me!

**Pull request merged**